### PR TITLE
NodeDef create hierarchy refactoring

### PR DESCRIPTION
**Pull request by sgiaccio**

### CODE REVIEW

File: server/modules/nodeDef/repository/nodeDefRepository.js [hunk 1]
```diff
@@ -31,23 +31,13 @@ const nodeDefSelectFields =
...
-  }
-
   return await client.one(`
         INSERT INTO ${getSurveyDBSchema(surveyId)}.node_def 
           (parent_uuid, uuid, type, props_draft, meta)
         VALUES ($1, $2, $3, $4, $5::jsonb)
         RETURNING *
     `,
-    [parentUuid, NodeDef.getUuid(nodeDef), NodeDef.getType(nodeDef), NodeDef.getProps(nodeDef), meta],
+    [parentUuid, NodeDef.getUuid(nodeDef), NodeDef.getType(nodeDef), NodeDef.getProps(nodeDef), JSON.stringify(NodeDef.getMeta(nodeDef))],
```
Comment (eliask) [inline server/modules/nodeDef/repository/nodeDefRepository.js]: Maybe the DB driver should do the conversion to JSON/JSONB?

Comment (fefrain) [reply server/modules/nodeDef/repository/nodeDefRepository.js]: Can you also inline parentUuid and avoid to write return...?

File: test/it/surveyTests/nodeDefTest.js [hunk 2]
```diff
@@ -12,14 +12,14 @@ const fetchRootNodeDef = async () => {
   return await NodeDefRepository.fetchRootNodeDef(Survey.getId(survey), true)
 }
 
-const createNodeDef = (parentNodeUuid, type, name) =>
-  NodeDef.newNodeDef(parentNodeUuid, type, Survey.cycleOneKey, {
+const createNodeDef = (parentNode, type, name) =>
```
Comment (fefrain) [inline test/it/surveyTests/nodeDefTest.js]: Can you rename parentNode to nodeDefParent to be consistent with our convention?

File: server/modules/nodeDef/repository/nodeDefRepository.js (hunk 1 above)
Comment (sgiaccio) [reply server/modules/nodeDef/repository/nodeDefRepository.js]: Ok

File: test/it/surveyTests/nodeDefTest.js (hunk 2 above)
Comment (sgiaccio) [reply test/it/surveyTests/nodeDefTest.js]: Ok

**Pull request merged**